src/near2far.cpp

```diff
@@ -334,6 +335,81 @@ void dft_near2far::save_farfields(const char *fname, const char *prefix, const v
   delete[] EH;
 }
 
+double *dft_near2far::flux(direction df, const volume &where, double resolution) {
+    /* compute output grid size etc. */
+    int dims[4] = {1,1,1,1};
+    double dx[3] = {0,0,0};
+    direction dirs[3] = {X,Y,Z};
+    int rank = 0, N = 1;
+    LOOP_OVER_DIRECTIONS(where.dim, d) {
+      dims[rank] = int(floor(where.in_direction(d) * resolution));
+      if (dims[rank] <= 1) {
+	dims[rank] = 1;
+	dx[rank] = 0;
+      }
+      else
+	dx[rank] = where.in_direction(d) / (dims[rank] - 1);
```

Comment:
You should accumulate the product of the nonzero `dx` elements and multiply this by the final flux at the end in order to get the integral.